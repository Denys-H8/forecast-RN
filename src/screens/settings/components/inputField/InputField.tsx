import React from 'react';
import { View, TextInput, TextInputProps } from 'react-native';

import constants from 'src/constants';
import { styles } from './inputField.styles';

type IProps = {} & TextInputProps;

const InputField: React.FC<IProps> = ({ ...props }) => {
  return (
    <View style={styles.inputContainer}>
      <TextInput
        placeholderTextColor={constants.SUB_COLOR}
        style={styles.input}
        {...props}
        testID={'InputField.TextInput'}
      />
    </View>
  );
};

export default InputField;
