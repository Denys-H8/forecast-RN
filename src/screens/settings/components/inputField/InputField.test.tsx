import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import InputField from './InputField';

const textValue = 'Some text value';
const newTextValue = 'New text value';
const placeholderValue = 'Placeholder value';

describe('InputField ', () => {
  test('should render InputField', () => {
    const onChangeTextMock = jest.fn();

    const { toJSON, getByTestId } = render(
      <InputField onChangeText={onChangeTextMock} placeholder={placeholderValue} value={textValue} />,
    );

    fireEvent.changeText(getByTestId('InputField.TextInput'), newTextValue);

    expect(toJSON()).toMatchSnapshot();
    expect(onChangeTextMock).toHaveBeenCalled();
    expect(onChangeTextMock).toHaveBeenCalledWith(newTextValue);
    expect(onChangeTextMock).not.toHaveBeenCalledWith('Some other value');
  });
});
