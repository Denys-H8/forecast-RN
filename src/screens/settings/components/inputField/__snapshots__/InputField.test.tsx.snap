// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField  should render InputField 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#191C39",
      "borderBottomColor": "#fff",
      "borderBottomWidth": 2,
      "marginBottom": 26,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    onChangeText={
      [MockFunction] {
        "calls": Array [
          Array [
            "New text value",
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    placeholder="Placeholder value"
    placeholderTextColor="#8B8EA5"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#fff",
        "fontFamily": "WorkSans-Regular.ttf",
        "fontSize": 16,
        "lineHeight": 18,
      }
    }
    testID="InputField.TextInput"
    underlineColorAndroid="transparent"
    value="Some text value"
  />
</View>
`;
